### DISCUSSION TITLE
Why not run foxglove websocket player in a worker？

### DISCUSSION THREAD

**Discussion by lizhucheng** [foxglove/community#4] [Q&A]
When I use foxglove websocket player to play a realtime protobuf msg stream，It delays after some minutes。But when I play a record include more data which is loaded in the memory，It looks ok。the different is how data has been loaded. Because websocket messages is handled in promise way in the same window, it blocks the ui.

**Comment by jtbandes** [comment]
Can you share details of the setup, e.g. what rate are you sending messages and how large are they? How much CPU/RAM does your computer have? What types of panels are you using in Studio?
